commit f2ab95814103314af3239d322e382c61c69a788d
Author: David S. Miller <davem@davemloft.net>
Date:   Sun Dec 23 16:01:35 2018 -0800

    net: Revert recent Spectre-v1 patches.
    
    This reverts:
    
    50d5258634ae ("net: core: Fix Spectre v1 vulnerability")
    d686026b1e6e ("phonet: af_phonet: Fix Spectre v1 vulnerability")
    a95386f0390a ("nfc: af_nfc: Fix Spectre v1 vulnerability")
    a3ac5817ffe8 ("can: af_can: Fix Spectre v1 vulnerability")
    
    After some discussion with Alexei Starovoitov these all seem to
    be completely unnecessary.
    
    Signed-off-by: David S. Miller <davem@davemloft.net>

diff --git a/net/phonet/af_phonet.c b/net/phonet/af_phonet.c
index d4b2abd78858..3b0ef691f5b1 100644
--- a/net/phonet/af_phonet.c
+++ b/net/phonet/af_phonet.c
@@ -34,8 +34,6 @@
 #include <net/phonet/phonet.h>
 #include <net/phonet/pn_dev.h>
 
-#include <linux/nospec.h>
-
 /* Transport protocol registration */
 static const struct phonet_protocol *proto_tab[PHONET_NPROTO] __read_mostly;
 
@@ -45,7 +43,6 @@ static const struct phonet_protocol *phonet_proto_get(unsigned int protocol)
 
 	if (protocol >= PHONET_NPROTO)
 		return NULL;
-	protocol = array_index_nospec(protocol, PHONET_NPROTO);
 
 	rcu_read_lock();
 	pp = rcu_dereference(proto_tab[protocol]);